### PR TITLE
introduce fdroid flavor

### PR DESCRIPTION
**Pull request by thermatk**
The heavy lifting for #136 is done
With this merged one can quickly release to F-Droid :+1:

### CODE REVIEW

**Comment by thermatk** [comment]
Please test that it didn't break the release config for what is now called the "gms" flavor, particularly, the `gmsRelease` variant that I can't test

File: app/build.gradle [hunk 1]
```diff
@@ -45,6 +44,18 @@ android {
...
 
+    productFlavors {
+        // includes gms
+        gms {
+            buildConfigField "boolean", "fdroid", "false"
+        }
+
+        // only foss
+        fdroid {
+            buildConfigField "boolean", "fdroid", "true"
```
Comment (k0shk0sh) [inline app/build.gradle]: can we make this capital, since its static final field

**Comment by thermatk** [comment]
F-Droid doesn't accept binary blobs and non-free/non-opensource code. The ads and billing are done using the gms, which I had to cut out
The flavor approach permits to replace them with something else, that's what I would expect. A link to support using bitcoins, paypal, patreon or I don't know :+1:

**Comment by thermatk** [comment]
https://f-droid.org/wiki/page/Antifeature:Ads

> No such apps are currently known to exist.

**Comment by k0shk0sh** [comment]
I didn't know that tbh as I have never published an app there. but good to know now :p anyway the ads were disabled by default so I guess there shouldn't be any prob.

**Comment by thermatk** [comment]
Only the API keys problem will remain, the easiest being for you to include some (probably separate?) with the fdroid flavor. I can include mine here, or maybe they can be applied during the build process on the f-droid servers as well

**Comment by k0shk0sh** [comment]
Ooops, I see that you removed `com.google.firebase.firebase-crash` which is needed to upload mapped proguard files to firebase.

**Comment by thermatk** [comment]
@k0shk0sh as a plugin? I couldn't find a reference to a crash plugin

**Comment by thermatk** [comment]
#296

**Pull request merged**